### PR TITLE
Make PID classes ready for ROOT6

### PR DESCRIPTION
**Pull request by MartinSchmidtTuebingen**
Make PID classes (AliTPCcalibResidualPID, TPCPIDEtaQA, TPCPIDEtaTree) and its macros ROOT6 ready, small code improvements

### CODE REVIEW

File: PWGPP/TPC/AliTPCcalibResidualPID.cxx [hunk 1]
```diff
@@ -3241,16 +3241,28 @@ TObjArray * AliTPCcalibResidualPID::GetResponseFunctions(TF1* parametrisation, T
...
-      graphKaonTPC->Fit(funcCorrKaon2, "QREX0M", "same", (isMC ? 1.981 : 1.0), 1.981);
+        TF1 * funcCorrKaon2 = new TF1("funcCorrKaon2", fitFuncString2.Data(), 1.0,  1.981);
+        graphKaonTPC->Fit(funcCorrKaon2, "QREX0M", "same", 1.0, 1.981);
 
-      funcCorrKaon = new TF1("funcCorrKaon",
-                             "funcCorrKaon1 * 0.5*(1.+TMath::Erf((1 - x) / 0.1)) + ([0]+[1]*x+[2]*x*x+[3]*x*x*x+[4]*x*x*x*x+[5]*x*x*x*x*x+[6]*x*x*x*x*x*x) * 0.5*(1.+TMath::Erf((x - 1) / 0.1))",
-                             TMath::Max(graphKaonTPC->GetX()[0], 0.25), 1.981);
+        funcCorrKaon = new TF1("funcCorrKaon",
+                                (funcCorrKaonString + TString(" * 0.5*(1.+TMath::Erf((1 - x) / 0.1)) + ([0]+[1]*x+[2]*x**2+[3]*x**3+[4]*x**4+[5]*x**5+[6]*x**6) * 0.5*(1.+TMath::Erf((x - 1) / 0.1))")).Data(),
+                                TMath::Max(graphKaonTPC->GetX()[0], 0.25), 1.981);
```
Comment (wiechula) [inline PWGPP/TPC/AliTPCcalibResidualPID.cxx:3265]: Are the modifications in this function specific for a certain period / production? Will they impact other spline productions?

File: PWGPP/TPC/macros/AddTaskTPCPIDEtaQA.C [hunk 2]
```diff
@@ -86,21 +82,23 @@ AliAnalysisTask *AddTaskTPCPIDEtaQA(TString period = "", Bool_t isPbpOrpPb = kFA
...
   //define output containers, please use 'username'_'somename'
   AliAnalysisDataContainer *coutput1 = 
       mgr->CreateContainer("TPCPIDEtaQA", TObjArray::Class(),
                            AliAnalysisManager::kOutputContainer,"TPCPIDEtaQA.root");
 
   //connect containers
   mgr->ConnectInput  (task,  0, cinput );
-  mgr->ConnectOutput (task,  0, coutput0);
+  
+  if (mgr->GetCommonOutputContainer()) {
```
Comment (wiechula) [inline PWGPP/TPC/macros/AddTaskTPCPIDEtaQA.C:93]: Why is this protection necessary?

**Review by wiechula — COMMENTED** [review]
@MartinSchmidtTuebingen , looks like mainly a restructuring of code. I guess the relevant change for ROOT6 is the enum?
A few comments / questions.

File: PWGPP/TPC/AliTPCcalibResidualPID.cxx [hunk 3]
```diff
@@ -3241,16 +3241,28 @@ TObjArray * AliTPCcalibResidualPID::GetResponseFunctions(TF1* parametrisation, T
...
-                                    TMath::Max(graphKaonTPC->GetX()[0], 0.25), 1.981); 
-      graphKaonTPC->Fit(funcCorrKaon1, "QREX0M", "same", TMath::Max(graphKaonTPC->GetX()[0], 0.25), 1.0);
+      //Simple functions
+//          Double_t minValue = TMath::Max(graphKaonTPC->GetX()[0], 0.25);
+//          Double_t maxValue = 1.0; //1.981
+//         funcCorrKaon = new TF1("funcCorrKaon", fitFuncString2.Data(), 0.22,  1.981);
+//         funcCorrKaon = new TF1("funcCorrKaon", "[0] + [1] / x + [2] / (x**2) + [3] / (x**3) + [4] / (x**4) + [5] / (x**5) + [6] * x + [7] / (x**6)", 0.22,  1.981);
+//          funcCorrKaon = new TF1("funcCorrKaon", fitFuncString.Data(), minValue,  maxValue);
+         
+//          graphKaonTPC->Fit(funcCorrKaon, "QREX0M", "same", minValue, maxValue);
```
[the remark this answers is not in the document]
Comment (MartinSchmidtTuebingen) [reply PWGPP/TPC/AliTPCcalibResidualPID.cxx:3251]: I used them as demonstration value and it is possible to fall back to them, but I can also remove them.

File: PWGPP/TPC/AliTPCcalibResidualPID.cxx (hunk 1 above)
Comment (MartinSchmidtTuebingen) [reply PWGPP/TPC/AliTPCcalibResidualPID.cxx:3265]: It is basically the same as before. However, ROOT6 cannot use simply the name of another function in the string to evaluate it, so I have to write it like this.

File: PWGPP/TPC/macros/AddTaskTPCPIDEtaQA.C (hunk 2 above)
Comment (MartinSchmidtTuebingen) [reply PWGPP/TPC/macros/AddTaskTPCPIDEtaQA.C:93]: Local trains dont have a CommonOutputContainer, so this fails in testing it locally.

File: PWGPP/TPC/AliTPCPIDEtaQA.cxx [hunk 4]
```diff
@@ -647,3 +627,23 @@ void AliTPCPIDEtaQA::SetUpHist(THnSparse* hist, Double_t* binsPt) const
...
+    //TODO fPhiCutLow  = new TF1("StandardPhiCutLow",  "0.1/x/x+pi/18.0-0.025", 0, 50);
+    //TODO fPhiCutHigh = new TF1("StandardPhiCutHigh", "0.12/x+pi/18.0+0.035", 0, 50);
+
+    //TODO NEW fPhiCutLow  = new TF1("StandardPhiCutLow",  "0.072/x+pi/18.0-0.035", 0, 50);
+    //TODO NEW fPhiCutHigh = new TF1("StandardPhiCutHigh", "0.07/x/x+0.1/x+pi/18.0+0.035", 0, 50);
+
+    // TODO fPhiCutSecondBranchLow has incorrect formula, but is not needed anyway at the moment
+//     fPhiCutSecondBranchLow  = new TF1("StandardPhiCutSecondBranchLow",  "StandardPhiCutLow - 2.*pi/18.", 0, 50);
+//     fPhiCutSecondBranchHigh = new TF1("StandardPhiCutSecondBranchHigh", "0.07/x+pi/18.0+0.125 - 2.*pi/18.", 0, 50);
+}
```
[the remark this answers is not in the document]
Comment (MartinSchmidtTuebingen) [reply PWGPP/TPC/AliTPCPIDEtaQA.cxx:647]: They are already initialised with a nullptr in the initialization list.

**Comment by pzhristov** [comment]
+test

**Comment by wiechula** [comment]
+1

**Pull request merged**